---
title: "SARS-CoV-2 vs RSV vs HPIV3: Enrichment of CARNIVAL results"
author: "Alberto Valdeolivas: alberto.valdeolivas@bioquant.uni-heidelberg.de; Date: "
date: "04/05/2020"
output: github_document
always_allow_html: true
---

```{r setup, include=FALSE, warning=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

### License Info

This program is free software: you can redistribute it and/or modify it under 
the terms of the GNU General Public License as published by the Free Software 
Foundation, either version 3 of the License, or (at your option) any later 
version.

This program is distributed in the hope that it will be useful, but WITHOUT ANY 
WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR 
A PARTICULAR PURPOSE. See the GNU General Public License for more details.

Please check http://www.gnu.org/licenses/.

## Introduction

The present script takes the RNAseq data from the study *"SARS-CoV-2 launches* 
*a unique transcriptional signature from in vitro, ex vivo, and in vivo systems"* 

<https://www.biorxiv.org/content/10.1101/2020.03.24.004655v1>

<https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE147507>

The goal of this set of scripts is to compare the transcriptional effect of 
different viral infections: SARS-CoV-2, Respiratory syncytial virus (RSV) and 
Human parainfluenza virus type 3 (HPIV3). In this script, we use the nodes from 
the **CARNIVAL** output to run an enrichment analysis for the conditions under 
study:

+ A549 alveolar cancer cell line: mock treated vs infected with SARS-CoV-2.   

+ A549 alveolar cancer cell line: mock treated vs infected with RSV. 

+ A549 alveolar cancer cell line: mock treated vs infected with HPIV3.

## Reading input data for Enrichment Analysis

To perform the enrichment analysis, we need to read the following input files:

+ Output from CARNIVAL: to obtain the significant genes and the background 
genes

+ Datasets from MSigDB: describing the pathways in which our significant genes
are known to be involved in. 

+ Differential expression analysis: To evaluate the activity of the genes 
involved in the different enriched pathaways.

We first load the required packages and we define some functions.

```{r, message=FALSE, warning=FALSE}
library(readr)
library(piano)
library(dplyr)
library(ggplot2)
library(omicToolsTest)
library(limma)
library(kableExtra)
library(tidyr)
library(ggsci)

## Function to extract the nodes that appear in CARNIVAL network and the 
## background genes (all genes present in the prior knowledge network).
## It returns a list with two objects: the success and the background genes.
extractCARNIVALnodes <- function(CarnivalResults){

    CarnivalNetwork <- 
        as.data.frame(CarnivalResults$weightedSIF, stringsAsFactors = FALSE)
    
    colnames(CarnivalNetwork) <- c("source", "sign", "target", "Weight")

    ## We define the set of nodes interesting for our condition
    sucesses <- unique(c(gsub("_.*","",CarnivalNetwork$source), 
        gsub("_.*","",CarnivalNetwork$target)))


    CarnivalAttributes <- as.data.frame(CarnivalResults$nodesAttributes, 
        stringsAsFactors = FALSE)

    ## We define the background as all the genes in our prior knowledge network.
    bg <- unique(gsub("_.*","",CarnivalAttributes$Node))     
    
    return(list(sucesses = sucesses, bg= bg))
}

### Function to print a barplot with the enriched pathways.
BarplotEnrichment <- function(PathwaysSelect, Interesting_pathways){ 
    
    p <- ggplot(PathwaysSelect, aes(x = reorder(pathway, pvalue), 
            y = -log10(pvalue))) + 
        geom_bar(aes(fill = sign), stat = "identity") +
        scale_fill_gradient2(low = "darkblue", high = "indianred", 
            mid = "whitesmoke", midpoint = 0) + 
        theme_minimal() +
        theme(axis.text.x = element_text(angle = 90, hjust = 1, 
            colour = ifelse(levels(reorder(PathwaysSelect$pathway, 
                PathwaysSelect$pvalue)) %in% Interesting_pathways, 
                "red", "grey40"),
            face = ifelse(levels(reorder(PathwaysSelect$pathway, 
                PathwaysSelect$pvalue)) %in% Interesting_pathways, 
                "bold", "plain")),
        panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank()) + 
        xlab("")
    return(p)
}    


### We modify the runGSAhyper from Piano to make it return the odds ratio and
### the confident level. 
runGSAhyper_V2 <- function (genes, pvalues, pcutoff, universe, gsc, gsSizeLim = c(1, 
    Inf), adjMethod = "fdr") 
{
    if (length(gsSizeLim) != 2) 
        stop("argument gsSizeLim should be a vector of length 2")
    if (missing(genes)) {
        stop("argument genes is required")
    }
    else {
        genes <- as.vector(as.matrix(genes))
        if (!is(genes, "character")) 
            stop("argument genes should be a character vector")
        if (length(unique(genes)) != length(genes)) 
            stop("argument genes should contain no duplicated entries")
    }
    if (missing(pvalues)) {
        pvalues <- rep(0, length(genes))
    }
    else {
        pvalues <- as.vector(as.matrix(pvalues))
        if (!is(pvalues, "numeric")) 
            stop("argument pvalues should be a numeric vector")
        if (length(pvalues) != length(genes)) 
            stop("argument pvalues should be the same length as argument genes")
        if (max(pvalues) > 1 | min(pvalues) < 0) 
            stop("pvalues need to lie between 0 and 1")
    }
    if (missing(pcutoff)) {
        if (all(pvalues %in% c(0, 1))) {
            pcutoff <- 0
        }
        else {
            pcutoff <- 0.05
        }
    }
    else {
        if (length(pcutoff) != 1 & !is(pcutoff, "numeric")) 
            stop("argument pcutoff should be a numeric of length 1")
        if (max(pcutoff) > 1 | min(pcutoff) < 0) 
            stop("argument pcutoff needs to lie between 0 and 1")
    }
    if (missing(gsc)) {
        stop("argument gsc needs to be given")
    }
    else {
        if (!is(gsc, "GSC")) 
            stop("argument gsc should be of class GSC, as returned by the loadGSC function")
    }
    if (missing(universe)) {
        if (!all(pvalues == 0)) {
            universe <- genes
            message("Using all genes in argument genes as universe.")
        }
        else {
            universe <- unique(unlist(gsc$gsc))
            message("Using all genes present in argument gsc as universe.")
        }
    }
    else {
        if (!is(universe, "character")) 
            stop("argument universe should be a character vector")
        if (!all(pvalues == 0)) 
            stop("if universe is given, genes should be only the genes of interest, i.e. pvalues should all be set to 0.")
    }
    if (!all(unique(unlist(gsc$gsc)) %in% universe)) 
        warning("there are genes in gsc that are not in the universe, these will be removed before analysis")
    if (!all(genes %in% universe)) {
        warning("not all genes given by argument genes are present in universe, these will be added to universe")
        universe <- c(universe, genes[!genes %in% universe])
    }
    if (length(unique(universe)) != length(universe)) 
        stop("argument universe should contain no duplicated entries")
    tmp <- try(adjMethod <- match.arg(adjMethod, c("holm", "hochberg", 
        "hommel", "bonferroni", "BH", "BY", "fdr", "none"), 
        several.ok = FALSE), silent = TRUE)
    if (is(tmp, "try-error")) {
        stop("argument adjMethod set to unknown method")
    }
    pvalues[pvalues == 0] <- -1e-10
    goi <- genes[pvalues < pcutoff]
    if (length(goi) < 1) 
        stop("no genes selected due to too strict pcutoff")
    bg <- universe[!universe %in% goi]
    gsc <- gsc$gsc
    delInd <- vector()
    for (i in 1:length(gsc)) {
        gs <- gsc[[i]]
        gs <- gs[gs %in% universe]
        if (length(gs) < gsSizeLim[1] | length(gs) > gsSizeLim[2]) 
            delInd <- c(delInd, i)
        gsc[[i]] <- gs
    }
    gsc <- gsc[!c(1:length(gsc)) %in% delInd]
    message(paste("Analyzing the overrepresentation of ", length(goi), 
        " genes of interest in ", length(gsc), " gene sets, using a background of ", 
        length(bg), " non-interesting genes.", sep = ""))
    p <- rep(NA, length(gsc))
    names(p) <- names(gsc)
    estimate <- rep(NA, length(gsc))
    names(estimate) <- names(gsc) 
    Cinterval <- rep(NA, length(gsc))
    names(Cinterval) <- names(gsc) 
    padj <- rep(NA, length(gsc))
    names(padj) <- names(gsc)
    contTabList <- list()
    resTab <- matrix(nrow = length(gsc), ncol = 8)
    colnames(resTab) <- c("p-value", "Adjusted p-value", "Significant (in gene set)", 
        "Non-significant (in gene set)", "Significant (not in gene set)", 
        "Non-significant (not in gene set)", "OddsRatio","ConfidentInterval")
    rownames(resTab) <- names(gsc)
    for (i in 1:length(gsc)) {
        gs <- gsc[[i]]
        nogs <- universe[!universe %in% gs]
        ctab <- rbind(c(sum(goi %in% gs), sum(goi %in% nogs)), 
            c(sum(bg %in% gs), sum(bg %in% nogs)))
        resultsfisher <- fisher.test(ctab, alternative = "greater")
        p[i] <- resultsfisher$p.value
        estimate[i] <- resultsfisher$estimate
        Cinterval[i] <- resultsfisher$conf.int[1]
        rownames(ctab) <- c("Significant", "Non-significant")
        colnames(ctab) <- c("Genes in gene set", "Genes not in gene set")
        contTabList[[i]] <- ctab
        resTab[i, ] <- c(p[i], NA, sum(goi %in% gs), sum(bg %in% 
            gs), sum(goi %in% nogs), sum(bg %in% nogs), estimate[i], Cinterval[i])
    }
    padj <- p.adjust(p, method = adjMethod)
    resTab[, 2] <- padj
    res <- list()
    res$pvalues <- p
    res$p.adj <- padj
    res$resTab <- resTab
    res$contingencyTable <- contTabList
    res$gsc <- gsc
    return(res)
}

```

### Reading and formatting CARNIVAL output

We read the CARNIVAL results generated in the previous script. We define
two different gene sets in order tor conduct the enrichment. The first set 
contains the nodes that appear in the CARNIVAL output and are therefore 
relevant in the context of our input transcriptomic data. The second set 
contains all the genes in our prior knowledge network which are used as the
backgroud. 

```{r, message=FALSE}
## SARS-CoV-2
CarnivalResultsA549vsCOV2_noinput <- 
    readRDS("ResultsCARNIVAL/A549vsCOV2_noinput.rds")
NodesA549vsCOV2_noinput <- extractCARNIVALnodes(CarnivalResultsA549vsCOV2_noinput)

CarnivalResultsA549vsCOV2_RIGIlike_receptors_input <- 
    readRDS("ResultsCARNIVAL/A549vsCOV2_RIGIlike_receptors_input.rds")
NodesA549vsCOV2_RIGIlike_receptors_input <- 
    extractCARNIVALnodes(CarnivalResultsA549vsCOV2_RIGIlike_receptors_input)  

## RSV
CarnivalResultsA549vsRSV_noinput <- 
    readRDS("ResultsCARNIVAL/A549vsRSV_noinput.rds")
NodesA549vsRSV_noinput <- extractCARNIVALnodes(CarnivalResultsA549vsRSV_noinput)

CarnivalResultsA549vsRSV_RIGIlike_receptors_input <- 
    readRDS("ResultsCARNIVAL/A549vsRSV_RIGIlike_receptors_input.rds")
NodesA549vsRSV_RIGIlike_receptors_input <- 
    extractCARNIVALnodes(CarnivalResultsA549vsRSV_RIGIlike_receptors_input) 

## SARS-CoV-2
CarnivalResultsA549vsHPIV3_noinput <- 
    readRDS("ResultsCARNIVAL/A549vsHPIV3_noinput.rds")
NodesA549vsHPIV3_noinput <- extractCARNIVALnodes(CarnivalResultsA549vsHPIV3_noinput)

CarnivalResultsA549vsHPIV3_RIGIlike_receptors_input <- 
    readRDS("ResultsCARNIVAL/A549vsHPIV3_RIGIlike_receptors_input.rds")
NodesA549vsHPIV3_RIGIlike_receptors_input <- 
    extractCARNIVALnodes(CarnivalResultsA549vsHPIV3_RIGIlike_receptors_input)
```

### Reading Pathway data sets from MSigDB

We downloaded from MSigDB <https://www.gsea-msigdb.org/> the following dataset:
c2.cp.v7.0.symbols.gmt. It contains several pathways from different resources
and the genes that are known to be involved in those pathways. 

```{r, message=FALSE, echo=TRUE}
pathways <- gmt_to_csv("../RawData/c2.cp.v7.0.symbols.gmt")
```

### Reading and formatting statistic from DEG

We read the results from the differential expression analysis. The statistic 
of the genes will be mapped later on in the different significant pathways. 

```{r, message=FALSE, warning=FALSE}
## Differential expression table
dds_A549vsCOV2 <- readRDS("IntermediateFiles/dds_results_A549vsCOV2.rds") %>%
    as.data.frame() %>% 
    dplyr::select(stat)
dds_A549vsRSV <- readRDS("IntermediateFiles/dds_results_A549vsRSV.rds") %>%
    as.data.frame() %>% 
    dplyr::select(stat)
dds_A549vsHPIV3 <- readRDS("IntermediateFiles/dds_results_A549vsHPIV3.rds") %>%
    as.data.frame() %>% 
    dplyr::select(stat)
```

## Performing Enrichment Analysis and plotting the Results

Using the **Piano** R package, we run a gene set analysis (GSA) based on a list 
of significant genes (CARNIVAL nodes) and a gene set collection (background). 
It uses Fisher's exact test. 

### CARNIVAL output with no perturbation

#### SARS-CoV-2 infection

```{r, message=FALSE, warning=FALSE}
## We run GSA hyper Geometric test
sig_pathways_A549vsCOV2_noinput <- runGSAhyper_V2(NodesA549vsCOV2_noinput$sucesses, 
    universe = NodesA549vsCOV2_noinput$bg, gsc = loadGSC(pathways))
sig_pathways_df_A549vsCOV2_noinput <- 
    as.data.frame(sig_pathways_A549vsCOV2_noinput$resTab)

## We map the t-stastic into the resulted enriched pathways.
sig_pathways_df_A549vsCOV2_noinput$sign <- 
    unlist(lapply(row.names(sig_pathways_df_A549vsCOV2_noinput), 
    function(x, kinases, pathways){
        return(mean(dds_A549vsCOV2[row.names(dds_A549vsCOV2) %in% pathways[pathways$term == x,1],1], na.rm = TRUE))
    },kinases = kinases, pathways = pathways))

sig_pathways_df_A549vsCOV2_noinput <- 
    sig_pathways_df_A549vsCOV2_noinput[!is.nan(sig_pathways_df_A549vsCOV2_noinput$sign),]
```

We format the results and we prepare them to be plotted. For visualization 
purposes, we just select pathways with adjusted p-values lower than 0.0001.

```{r, message=FALSE}
PathwaysSelect_A549vsCOV2_noinput <- sig_pathways_df_A549vsCOV2_noinput %>%
    tibble::rownames_to_column(var = "pathway") %>%
    dplyr::select(pathway, `p-value`, `Adjusted p-value`, sign) %>%
    dplyr::filter(`Adjusted p-value` <= 0.0001) %>%
    dplyr::rename(pvalue = `p-value`, AdjPvalu = `Adjusted p-value`) %>% 
    dplyr::mutate(pathway = as.factor(pathway))
```

We finally plot the results highlighting the most relevant pathways.

```{r, message=FALSE, warning=FALSE}
Interesting_pathways_A549vsCOV2_noinput <- c()

p_A549vsCOV2_noinput <- BarplotEnrichment(PathwaysSelect_A549vsCOV2_noinput, 
    Interesting_pathways_A549vsCOV2_noinput)
```

```{r, fig.width=12, fig.height=12, echo=FALSE, dip=300}
p_A549vsCOV2_noinput
```

#### RSV infection

```{r, message=FALSE, warning=FALSE}
## We run GSA hyper Geometric test
sig_pathways_A549vsRSV_noinput <- runGSAhyper_V2(NodesA549vsRSV_noinput$sucesses, 
    universe = NodesA549vsRSV_noinput$bg, gsc = loadGSC(pathways))
sig_pathways_df_A549vsRSV_noinput <- 
    as.data.frame(sig_pathways_A549vsRSV_noinput$resTab)

## We map the t-stastic into the resulted enriched pathways.
sig_pathways_df_A549vsRSV_noinput$sign <- 
    unlist(lapply(row.names(sig_pathways_df_A549vsRSV_noinput), 
    function(x, kinases, pathways){
        return(mean(dds_A549vsRSV[row.names(dds_A549vsRSV) %in% pathways[pathways$term == x,1],1], na.rm = TRUE))
    },kinases = kinases, pathways = pathways))

sig_pathways_df_A549vsRSV_noinput <- 
    sig_pathways_df_A549vsRSV_noinput[!is.nan(sig_pathways_df_A549vsRSV_noinput$sign),]
```

We format the results and we prepare them to be plotted. For visualization 
purposes, we just select pathways with adjusted p-values lower than 0.0001.

```{r, message=FALSE}
PathwaysSelect_A549vsRSV_noinput <- sig_pathways_df_A549vsRSV_noinput %>%
    tibble::rownames_to_column(var = "pathway") %>%
    dplyr::select(pathway, `p-value`, `Adjusted p-value`, sign) %>%
    dplyr::filter(`Adjusted p-value` <= 0.0001) %>%
    dplyr::rename(pvalue = `p-value`, AdjPvalu = `Adjusted p-value`) %>% 
    dplyr::mutate(pathway = as.factor(pathway))
```

We finally plot the results highlighting the most relevant pathways.

```{r, message=FALSE, warning=FALSE}
Interesting_pathways_A549vsRSV_noinput <- c()

p_A549vsRSV_noinput <- BarplotEnrichment(PathwaysSelect_A549vsRSV_noinput, 
    Interesting_pathways_A549vsRSV_noinput)
```

```{r, fig.width=12, fig.height=12, echo=FALSE, dip=300}
p_A549vsRSV_noinput
```

#### HPIV3 infection

```{r, message=FALSE, warning=FALSE}
## We run GSA hyper Geometric test
sig_pathways_A549vsHPIV3_noinput <- runGSAhyper_V2(NodesA549vsHPIV3_noinput$sucesses, 
    universe = NodesA549vsHPIV3_noinput$bg, gsc = loadGSC(pathways))
sig_pathways_df_A549vsHPIV3_noinput <- 
    as.data.frame(sig_pathways_A549vsHPIV3_noinput$resTab)

## We map the t-stastic into the resulted enriched pathways.
sig_pathways_df_A549vsHPIV3_noinput$sign <- 
    unlist(lapply(row.names(sig_pathways_df_A549vsHPIV3_noinput), 
    function(x, kinases, pathways){
        return(mean(dds_A549vsHPIV3[row.names(dds_A549vsHPIV3) %in% pathways[pathways$term == x,1],1], na.rm = TRUE))
    },kinases = kinases, pathways = pathways))

sig_pathways_df_A549vsHPIV3_noinput <- 
    sig_pathways_df_A549vsHPIV3_noinput[!is.nan(sig_pathways_df_A549vsHPIV3_noinput$sign),]
```

We format the results and we prepare them to be plotted. For visualization 
purposes, we just select pathways with adjusted p-values lower than 0.025.

```{r, message=FALSE}
PathwaysSelect_A549vsHPIV3_noinput <- sig_pathways_df_A549vsHPIV3_noinput %>%
    tibble::rownames_to_column(var = "pathway") %>%
    dplyr::select(pathway, `p-value`, `Adjusted p-value`, sign) %>%
    dplyr::filter(`Adjusted p-value` <= 0.025) %>%
    dplyr::rename(pvalue = `p-value`, AdjPvalu = `Adjusted p-value`) %>% 
    dplyr::mutate(pathway = as.factor(pathway))
```

We finally plot the results highlighting the most relevant pathways.

```{r, message=FALSE, warning=FALSE}
Interesting_pathways_A549vsHPIV3_noinput <- c()

p_A549vsHPIV3_noinput <- BarplotEnrichment(PathwaysSelect_A549vsHPIV3_noinput, 
    Interesting_pathways_A549vsHPIV3_noinput)
```

```{r, fig.width=12, fig.height=12, echo=FALSE, dip=300}
p_A549vsHPIV3_noinput
```

### Pathways with different behaviour in SARS-CoV-2 and the other infections

We now compare the enrichment results of the SARS-CoV-2 VS the other two
viral infections (HPIV3 and RSV) in order to identify specific
deregulated pathways during SARS-CoV-2 infection. To do so, we use limma on the
odds ratios from the fisher's exact test carried out in the previous 
enrichments. We sort the results based on the larger difference in odds ratios 
between conditions. In addition, we only consider the pathways that are 
significantly enriched in any of the conditions (Adjusted p-value < 0.1). 

```{r, warning=FALSE, message=FALSE}
AllEnrichments_noinput <- bind_cols(
    sig_pathways_df_A549vsCOV2_noinput %>% 
        tibble::rownames_to_column(var = "Pathway") %>% 
        dplyr::rename(OddsRatio_COV2 = "OddsRatio") %>% 
        dplyr::select(Pathway, OddsRatio_COV2),
    sig_pathways_df_A549vsRSV_noinput  %>% 
        dplyr::rename(OddsRatio_RSV = "OddsRatio") %>% 
        dplyr::select(OddsRatio_RSV),
    sig_pathways_df_A549vsHPIV3_noinput  %>% 
        dplyr::rename(OddsRatio_HPIV3 = "OddsRatio") %>% 
        dplyr::select(OddsRatio_HPIV3)) %>% 
    dplyr::filter(!(OddsRatio_COV2 == 0 & OddsRatio_RSV == 0 & OddsRatio_HPIV3 == 0)) %>% 
    tibble::column_to_rownames(var = "Pathway")
# rownames(AllEnrichments_noinput) <- rownames(sig_pathways_df_A549vsCOV2_noinput)
design <- cbind(Grp1=1,Grp2vs1=c(1,-1,-1))

## Limma model
fit_noinput <- lmFit(AllEnrichments_noinput,design)
fit_noinput <- eBayes(fit_noinput)
Results_noinput <- topTable(fit_noinput, sort.by = "B",resort.by="AveExpr", 
    number = 100)

## We tidy up the results and we include the p-values from the enrichments.
Results_noinput_OddsRatio <- 
    AllEnrichments_noinput[rownames(Results_noinput),] %>% 
    tibble::rownames_to_column(var = "Pathway") %>%
    dplyr::left_join((sig_pathways_df_A549vsCOV2_noinput %>% 
        tibble::rownames_to_column(var = "Pathway"))) %>% 
    dplyr::select(Pathway, OddsRatio_COV2, OddsRatio_RSV, 
                  OddsRatio_HPIV3, `Adjusted p-value`, `p-value`) %>% 
    dplyr::rename(Adjpvalue_COV2 = "Adjusted p-value") %>% 
    dplyr::rename(Pvalue_COV2 = "p-value") %>% 
    dplyr::left_join((sig_pathways_df_A549vsRSV_noinput %>% 
        tibble::rownames_to_column(var = "Pathway"))) %>% 
    dplyr::select(Pathway, OddsRatio_COV2, OddsRatio_RSV, 
        OddsRatio_HPIV3, Adjpvalue_COV2, Pvalue_COV2, `Adjusted p-value`, `p-value`) %>% 
    dplyr::rename(Adjpvalue_RSV = "Adjusted p-value") %>% 
    dplyr::rename(Pvalue_RSV = "p-value") %>%
    dplyr::left_join((sig_pathways_df_A549vsHPIV3_noinput %>% 
        tibble::rownames_to_column(var = "Pathway"))) %>% 
    dplyr::select(Pathway, OddsRatio_COV2, OddsRatio_RSV, 
        OddsRatio_HPIV3, Adjpvalue_COV2, Pvalue_COV2, Adjpvalue_RSV , Pvalue_RSV,
        `Adjusted p-value`, `p-value`) %>% 
    dplyr::rename(Adjpvalue_HPIV3 = "Adjusted p-value") %>%
    dplyr::rename(Pvalue_HPIV3 = "p-value") %>%
    dplyr::filter(Adjpvalue_COV2 < 0.1 | (Adjpvalue_RSV < 0.1 & Adjpvalue_HPIV3 <0.1))
    
## We presenth the results in a table
knitr::kable(Results_noinput_OddsRatio,  digits = 3, longtable = TRUE, 
    padding = 0) %>% 
    kable_styling(bootstrap_options = 
        c("striped", "hover", "condensed", "responsive"),
        full_width = F, font_size = 8)
```

And we finally plot the results

```{r, message=FALSE}
df_to_plot_OddsRatios_noinput <- Results_noinput_OddsRatio %>% 
    pivot_longer(cols = c(OddsRatio_COV2, OddsRatio_RSV,
    OddsRatio_HPIV3), names_to = "Virus", values_to = "OddsRatio") %>% 
    dplyr::select(Pathway, Virus, OddsRatio) %>%
    dplyr::mutate(Virus = sub("OddsRatio_","", Virus))

df_to_plot_Pvalues_noinput <- Results_noinput_OddsRatio %>% 
    pivot_longer(cols = c(Pvalue_COV2, Pvalue_RSV,
    Pvalue_HPIV3), names_to = "Virus", values_to = "Pvalues") %>% 
    dplyr::select(Pathway, Virus, Pvalues) %>%
    dplyr::mutate(Virus = sub("Pvalue_","", Virus))

df_to_plot_noinput <- 
    left_join(df_to_plot_OddsRatios_noinput, df_to_plot_Pvalues_noinput)

PointPlot_noinput <- ggplot(df_to_plot_noinput, aes(Pathway, -log(Pvalues))) + 
    geom_point(aes(color = Virus, size = OddsRatio)) + 
    coord_flip() + 
    theme_minimal() + 
    theme(legend.position = "bottom",  legend.justification = "center") +
    scale_color_lancet() +
    theme(axis.text.x = element_text(hjust = 1, size =8),
        axis.text.y = element_text(size =8),
        panel.grid.minor = element_blank()) 
```

```{r, echo=FALSE, dip=300, fig.width=10}
PointPlot_noinput
```

### CARNIVAL output with perturbations on the RIG-I-like receptors

#### SARS-CoV-2 infection

```{r, message=FALSE, warning=FALSE}
## We run GSA hyper Geometric test
sig_pathways_A549vsCOV2_RIGIlike_receptors_input <- 
runGSAhyper_V2(NodesA549vsCOV2_RIGIlike_receptors_input$sucesses, 
    universe = NodesA549vsCOV2_RIGIlike_receptors_input$bg, 
    gsc = loadGSC(pathways))
sig_pathways_df_A549vsCOV2_RIGIlike_receptors_input <- 
    as.data.frame(sig_pathways_A549vsCOV2_RIGIlike_receptors_input$resTab)

## We map the t-stastic into the resulted enriched pathways.
sig_pathways_df_A549vsCOV2_RIGIlike_receptors_input$sign <- 
    unlist(lapply(row.names(sig_pathways_df_A549vsCOV2_RIGIlike_receptors_input), 
    function(x, kinases, pathways){
        return(mean(dds_A549vsCOV2[row.names(dds_A549vsCOV2) %in% pathways[pathways$term == x,1],1], na.rm = TRUE))
    },kinases = kinases, pathways = pathways))

sig_pathways_df_A549vsCOV2_RIGIlike_receptors_input <- 
    sig_pathways_df_A549vsCOV2_RIGIlike_receptors_input[!is.nan(sig_pathways_df_A549vsCOV2_RIGIlike_receptors_input$sign),]
```

We format the results and we prepare them to be plotted. For visualization 
purposes, we just select pathways with adjusted p-values lower than 0.0001.

```{r, message=FALSE}
PathwaysSelect_A549vsCOV2_RIGIlike_receptors_input <- 
    sig_pathways_df_A549vsCOV2_RIGIlike_receptors_input %>%
    tibble::rownames_to_column(var = "pathway") %>%
    dplyr::select(pathway, `p-value`, `Adjusted p-value`, sign) %>%
    dplyr::filter(`Adjusted p-value` <= 0.0001) %>%
    dplyr::rename(pvalue = `p-value`, AdjPvalu = `Adjusted p-value`) %>% 
    dplyr::mutate(pathway = as.factor(pathway))
```

We finally plot the results highlighting the most relevant pathways.

```{r, message=FALSE, warning=FALSE}
Interesting_pathways_A549vsCOV2_RIGIlike_receptors_input <- c()

p_A549vsCOV2_RIGIlike_receptors_input <- 
    BarplotEnrichment(PathwaysSelect_A549vsCOV2_RIGIlike_receptors_input, 
    Interesting_pathways_A549vsCOV2_RIGIlike_receptors_input)
```

```{r, fig.width=12, fig.height=12, echo=FALSE, dip=300}
p_A549vsCOV2_RIGIlike_receptors_input
```

#### RSV infection

```{r, message=FALSE, warning=FALSE}
## We run GSA hyper Geometric test
sig_pathways_A549vsRSV_RIGIlike_receptors_input <- 
    runGSAhyper_V2(NodesA549vsRSV_RIGIlike_receptors_input$sucesses, 
    universe = NodesA549vsRSV_RIGIlike_receptors_input$bg, 
    gsc = loadGSC(pathways))
sig_pathways_df_A549vsRSV_RIGIlike_receptors_input <- 
    as.data.frame(sig_pathways_A549vsRSV_RIGIlike_receptors_input$resTab)

## We map the t-stastic into the resulted enriched pathways.
sig_pathways_df_A549vsRSV_RIGIlike_receptors_input$sign <- 
    unlist(lapply(row.names(sig_pathways_df_A549vsRSV_RIGIlike_receptors_input), 
    function(x, kinases, pathways){
        return(mean(dds_A549vsRSV[row.names(dds_A549vsRSV) %in% pathways[pathways$term == x,1],1], na.rm = TRUE))
    },kinases = kinases, pathways = pathways))

sig_pathways_df_A549vsRSV_RIGIlike_receptors_input <- 
    sig_pathways_df_A549vsRSV_RIGIlike_receptors_input[!is.nan(sig_pathways_df_A549vsRSV_RIGIlike_receptors_input$sign),]
```

We format the results and we prepare them to be plotted. For visualization 
purposes, we just select pathways with adjusted p-values lower than 0.01.

```{r, message=FALSE}
PathwaysSelect_A549vsRSV_RIGIlike_receptors_input <- 
    sig_pathways_df_A549vsRSV_RIGIlike_receptors_input %>%
    tibble::rownames_to_column(var = "pathway") %>%
    dplyr::select(pathway, `p-value`, `Adjusted p-value`, sign) %>%
    dplyr::filter(`Adjusted p-value` <= 0.01) %>%
    dplyr::rename(pvalue = `p-value`, AdjPvalu = `Adjusted p-value`) %>% 
    dplyr::mutate(pathway = as.factor(pathway))
```

We finally plot the results highlighting the most relevant pathways.

```{r, message=FALSE, warning=FALSE}
Interesting_pathways_A549vsRSV_RIGIlike_receptors_input <- c()

p_A549vsRSV_RIGIlike_receptors_input <- 
    BarplotEnrichment(PathwaysSelect_A549vsRSV_RIGIlike_receptors_input, 
    Interesting_pathways_A549vsRSV_RIGIlike_receptors_input)
```

```{r, fig.width=12, fig.height=12, echo=FALSE, dip=300}
p_A549vsRSV_RIGIlike_receptors_input
```

#### HPIV3 infection

```{r, message=FALSE, warning=FALSE}
## We run GSA hyper Geometric test
sig_pathways_A549vsHPIV3_RIGIlike_receptors_input <- runGSAhyper_V2(NodesA549vsHPIV3_RIGIlike_receptors_input$sucesses, 
    universe = NodesA549vsHPIV3_RIGIlike_receptors_input$bg, 
    gsc = loadGSC(pathways))
sig_pathways_df_A549vsHPIV3_RIGIlike_receptors_input <- 
    as.data.frame(sig_pathways_A549vsHPIV3_RIGIlike_receptors_input$resTab)

## We map the t-stastic into the resulted enriched pathways.
sig_pathways_df_A549vsHPIV3_RIGIlike_receptors_input$sign <- 
    unlist(lapply(row.names(sig_pathways_df_A549vsHPIV3_RIGIlike_receptors_input), 
    function(x, kinases, pathways){
        return(mean(dds_A549vsHPIV3[row.names(dds_A549vsHPIV3) %in% pathways[pathways$term == x,1],1], na.rm = TRUE))
    },kinases = kinases, pathways = pathways))

sig_pathways_df_A549vsHPIV3_RIGIlike_receptors_input <- 
    sig_pathways_df_A549vsHPIV3_RIGIlike_receptors_input[!is.nan(sig_pathways_df_A549vsHPIV3_RIGIlike_receptors_input$sign),]
```

We format the results and we prepare them to be plotted. For visualization 
purposes, we just select pathways with adjusted p-values lower than 0.0005

```{r, message=FALSE}
PathwaysSelect_A549vsHPIV3_RIGIlike_receptors_input <- 
    sig_pathways_df_A549vsHPIV3_RIGIlike_receptors_input %>%
    tibble::rownames_to_column(var = "pathway") %>%
    dplyr::select(pathway, `p-value`, `Adjusted p-value`, sign) %>%
    dplyr::filter(`Adjusted p-value` <= 0.0005) %>%
    dplyr::rename(pvalue = `p-value`, AdjPvalu = `Adjusted p-value`) %>% 
    dplyr::mutate(pathway = as.factor(pathway))
```

We finally plot the results highlighting the most relevant pathways.

```{r, message=FALSE, warning=FALSE}
Interesting_pathways_A549vsHPIV3_RIGIlike_receptors_input <- c()

p_A549vsHPIV3_RIGIlike_receptors_input <- 
    BarplotEnrichment(PathwaysSelect_A549vsHPIV3_RIGIlike_receptors_input, 
    Interesting_pathways_A549vsHPIV3_RIGIlike_receptors_input)
```

```{r, fig.width=12, fig.height=12, echo=FALSE, dip=300}
p_A549vsHPIV3_RIGIlike_receptors_input
```

### Pathways with different behaviour in SARS-CoV-2 and the other infections

We now compare the enrichment results of the SARS-CoV-2 VS the other two
viral infections (HPIV3 and RSV) in order to identify specific
deregulated pathways during SARS-CoV-2 infection. To do so, we use limma on the
odds ratios from the fisher's exact test carried out in the previous 
enrichments. We sort the results based on the larger difference in odds ratios 
between conditions. In addition, we only consider the pathways that are 
significantly enriched in any of the conditions (Adjusted p-value < 0.1). 

```{r, warning=FALSE, message=FALSE}
AllEnrichments_RIGIlike_receptors_input <- bind_cols(
    sig_pathways_df_A549vsCOV2_RIGIlike_receptors_input %>% 
        tibble::rownames_to_column(var = "Pathway") %>% 
        dplyr::rename(OddsRatio_COV2 = "OddsRatio") %>% 
        dplyr::select(Pathway, OddsRatio_COV2),
    sig_pathways_df_A549vsRSV_RIGIlike_receptors_input  %>% 
        dplyr::rename(OddsRatio_RSV = "OddsRatio") %>% 
        dplyr::select(OddsRatio_RSV),
    sig_pathways_df_A549vsHPIV3_RIGIlike_receptors_input  %>% 
        dplyr::rename(OddsRatio_HPIV3 = "OddsRatio") %>% 
        dplyr::select(OddsRatio_HPIV3)) %>% 
    dplyr::filter(!(OddsRatio_COV2 == 0 & OddsRatio_RSV == 0 & OddsRatio_HPIV3 == 0)) %>% 
    tibble::column_to_rownames(var = "Pathway")
# rownames(AllEnrichments_RIGIlike_receptors_input) <- rownames(sig_pathways_df_A549vsCOV2_RIGIlike_receptors_input)
design <- cbind(Grp1=1,Grp2vs1=c(1,-1,-1))

## Limma model
fit_RIGIlike_receptors_input <- lmFit(AllEnrichments_RIGIlike_receptors_input,design)
fit_RIGIlike_receptors_input <- eBayes(fit_RIGIlike_receptors_input)
Results_RIGIlike_receptors_input <- topTable(fit_RIGIlike_receptors_input, 
    sort.by = "B",resort.by="AveExpr", number = 100)

## We tidy up the results and we include the p-values from the enrichments.
Results_RIGIlike_receptors_input_OddsRatio <- 
    AllEnrichments_RIGIlike_receptors_input[rownames(Results_RIGIlike_receptors_input),] %>% 
    tibble::rownames_to_column(var = "Pathway") %>%
    dplyr::left_join((sig_pathways_df_A549vsCOV2_RIGIlike_receptors_input %>% 
        tibble::rownames_to_column(var = "Pathway"))) %>% 
    dplyr::select(Pathway, OddsRatio_COV2, OddsRatio_RSV, 
                  OddsRatio_HPIV3, `Adjusted p-value`, `p-value`) %>% 
    dplyr::rename(Adjpvalue_COV2 = "Adjusted p-value") %>% 
    dplyr::rename(Pvalue_COV2 = "p-value") %>% 
    dplyr::left_join((sig_pathways_df_A549vsRSV_RIGIlike_receptors_input %>% 
        tibble::rownames_to_column(var = "Pathway"))) %>% 
    dplyr::select(Pathway, OddsRatio_COV2, OddsRatio_RSV, 
        OddsRatio_HPIV3, Adjpvalue_COV2, Pvalue_COV2, `Adjusted p-value`, `p-value`) %>% 
    dplyr::rename(Adjpvalue_RSV = "Adjusted p-value") %>% 
    dplyr::rename(Pvalue_RSV = "p-value") %>%
    dplyr::left_join((sig_pathways_df_A549vsHPIV3_RIGIlike_receptors_input %>% 
        tibble::rownames_to_column(var = "Pathway"))) %>% 
    dplyr::select(Pathway, OddsRatio_COV2, OddsRatio_RSV, 
        OddsRatio_HPIV3, Adjpvalue_COV2, Pvalue_COV2, Adjpvalue_RSV , Pvalue_RSV,
        `Adjusted p-value`, `p-value`) %>% 
    dplyr::rename(Adjpvalue_HPIV3 = "Adjusted p-value") %>%
    dplyr::rename(Pvalue_HPIV3 = "p-value") %>%
    dplyr::filter(Adjpvalue_COV2 < 0.1 | (Adjpvalue_RSV < 0.1 & Adjpvalue_HPIV3 <0.1))


## We presenth the results in a table
knitr::kable(Results_RIGIlike_receptors_input_OddsRatio,  digits = 3, longtable = TRUE, 
    padding = 0) %>% 
    kable_styling(bootstrap_options = 
        c("striped", "hover", "condensed", "responsive"),
        full_width = F, font_size = 8)
```

And we finally plot the results

```{r, message=FALSE}
df_to_plot_OddsRatios_RIGIlike_receptors_input <- Results_RIGIlike_receptors_input_OddsRatio %>% 
    pivot_longer(cols = c(OddsRatio_COV2, OddsRatio_RSV,
    OddsRatio_HPIV3), names_to = "Virus", values_to = "OddsRatio") %>% 
    dplyr::select(Pathway, Virus, OddsRatio) %>%
    dplyr::mutate(Virus = sub("OddsRatio_","", Virus))

df_to_plot_Pvalues_RIGIlike_receptors_input <- Results_RIGIlike_receptors_input_OddsRatio %>% 
    pivot_longer(cols = c(Pvalue_COV2, Pvalue_RSV,
    Pvalue_HPIV3), names_to = "Virus", values_to = "Pvalues") %>% 
    dplyr::select(Pathway, Virus, Pvalues) %>%
    dplyr::mutate(Virus = sub("Pvalue_","", Virus))

df_to_plot_RIGIlike_receptors_input <- 
    left_join(df_to_plot_OddsRatios_RIGIlike_receptors_input, 
              df_to_plot_Pvalues_RIGIlike_receptors_input)

## For now I am going to take out REACTOME_REGULATION_OF
df_to_plot_RIGIlike_receptors_input <-
    dplyr::filter(df_to_plot_RIGIlike_receptors_input, 
    Pathway != "REACTOME_REGULATION_OF_GENE_EXPRESSION_IN_LATE_STAGE_BRANCHING_MORPHOGENESIS_PANCREATIC_BUD_PRECURSOR_CELLS")

PointPlot_RIGIlike_receptors_input <- ggplot(df_to_plot_RIGIlike_receptors_input, aes(Pathway, -log(Pvalues))) + 
    geom_point(aes(color = Virus, size = OddsRatio)) + 
    coord_flip() + 
    theme_minimal() + 
    theme(legend.position = "bottom",  legend.justification = "center") +
    scale_color_lancet() +
    theme(axis.text.x = element_text(hjust = 1, size =8),
        axis.text.y = element_text(size =8),
        panel.grid.minor = element_blank()) 
```

```{r, echo=FALSE, dip=300, fig.width=10}
PointPlot_RIGIlike_receptors_input
```

## Session Info Details

```{r, echo=FALSE, eval=TRUE}
sessionInfo()
```

